backend/app/services/file_service.py

```
@@ -0,0 +1,80 @@
+    return suffix
+
+
+def validate_video_file(upload_file: UploadFile) -> str:
+    suffix = _safe_suffix(upload_file.filename)
+    if suffix not in ALLOWED_VIDEO_EXTENSIONS:
+        raise ValueError("Unsupported file extension. Allowed: MP4, AVI, MKV, MOV")
+
+    content_type = (upload_file.content_type or "").lower()
+    if content_type and content_type != "application/octet-stream":
+        if content_type not in ALLOWED_VIDEO_MIME_TYPES:
+            raise ValueError("Unsupported MIME type for video upload")
```

Ruff (0.15.4)

[warning] Avoid specifying long messages outside the exception class

(TRY003)